src/modules/audio/source.c

```diff
@@ -165,6 +165,12 @@ void lovrSourcePlay(Source* source) {
     return;
   }
 
+  // in case we have some queued buffers, make sure to unqueue them before streaming more data into them.
+  ALint processed;
+  alGetSourcei(lovrSourceGetId(source), AL_BUFFERS_PROCESSED, &processed);
+  ALuint buffers[SOURCE_BUFFERS];
+  alSourceUnqueueBuffers(source->id, processed, buffers);
+
   lovrSourceStream(source, source->buffers, SOURCE_BUFFERS);
```

Comment:
We can end up in play() AFTER all buffers are exhaused and processed, but BEFORE AudioUpdate has run. In this case, the OpenAL source is STOPPED (so the guard at start of play() lets us through), but some buffers are still processed and queued and need to unqueued.

This means that after we unqueue the last processed buffers, ALL buffersd are now unqueued and ready for reuse. So we can now use source->buffers and don't have to care about the set of unqueued buffers.

So this code is correct. I'm 99.9% sure :) and it was likely broken even before my changes for code that very aggressively tries to play() when source becomes STOPPED.

Reply:
Ok, thanks for the explanation :) a small nit would then be to use NULL for the buffers array (like we do in lovrSourceStop) so future people don't mistakenly assume it was a typo.

Reply:
so to clarify: the intent of both the original and this modified code is indeed that play() can't be called unless source was stopped, implicitly (by running out of audio to play) or explicitly! but stopped and processed doesn't mean unqueued; this is taken care of both by audio_update and this code, because there is no guarantee that update is called before someone calls play() on it..

also, waiting for audioupdate to do the queueing is bad for latency :)

Reply:
Ok, thanks for the explanation :) a small nit would then be to use NULL for the buffers array (like we do in lovrSourceStop) so future people don't mistakenly assume it was a typo.

Reply:
fixed and documented!